### PR TITLE
🌱  Run golanci-lint in parallel and enable more linters

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

This PR revamps how we use golanci-lint in our Makefile, to make sure it can run in parallel when we run `make lint`. It also removes the usage of `lint` and `lint-full`. We should always run all the linters when `make lint` is run.

There is also a new command `make lint-fix` which is going to ask some of the linters to automatically fix the issues they've found.

The maini reasoning to add the new linter is to make sure our comments, imports, and so on are more consistent going forward.

/milestone v0.4
/assign @CecileRobertMichon

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -195,16 +195,22 @@ e2e-framework: ## Builds the CAPI e2e framework
...
-	cd $(CAPD_DIR); $(GOLANGCI_LINT) run -v
-
-lint-full: $(GOLANGCI_LINT) ## Run slower linters to detect possible issues
-	$(GOLANGCI_LINT) run -v --fast=false
-	cd $(E2E_FRAMEWORK_DIR); $(GOLANGCI_LINT) run -v --fast=false
-	cd $(CAPD_DIR); $(GOLANGCI_LINT) run -v --fast=false
+	$(MAKE) -j8 lint-all
+
+.PHONY: lint-all lint-core lint-e2e lint-capd
+lint-all: lint-core lint-e2e lint-capd
```
[the remark this answers is not in the document]
Comment (vincepri) [reply Makefile:203]: Clusterctl is included in the core because it's not a go module

**Comment by CecileRobertMichon** [comment]
lgtm but the tests are failing

**Comment by CecileRobertMichon** [comment]
/hold

pull-cluster-api-verify seems to actually be taking longer now (13 minutes as opposed to 9 minutes before), are we sure the linters are running in parallel? Wonder if it's taking longer because of the additional linters of something else

**Comment by vincepri** [comment]
@CecileRobertMichon yeah the additional linter might be increasing the runtime. Locally it's definitely faster, but CI is a different story.

Next step though is to remove the ci-make target and add the github action which would give us inline errors as well when looking at the code diff.

**Comment by alexander-demichev** [comment]
A question not directly related to this PR, is there a linter that can forbid `v1` import of klog? I quickly went through linters and didn't find any.

**Comment by vincepri** [comment]
There is one, but it's not working, I haven't had time to dig in further

**Comment by CecileRobertMichon** [comment]
/hold cancel

lgtm pending rebase

**Comment by CecileRobertMichon** [comment]
race condition

/lgtm
/approve

**Comment by vincepri** [comment]
Rebased :)

**Pull request merged**